### DISCUSSION TITLE
Suche in Spalte Typ Date - Error 400

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#71] [General]
Hallo Jürgen!

Wir haben eine Suche nach dem Erhebungsdatum nach/vor konfiguriert (Abfragemethode GreaterThan bzw. LowerThan).
Die Spalte ist vom Type Date und es ist ein ArcGIS Pro Map+Feature-Service.

Beim Ausführen der Suche kommt ein Error 400
![error_webgis](https://github.com/gis-eni/webgis-forum/assets/122986367/7b8416e3-d307-47f6-b929-8da9da9f973d)

Im AGS Logfile ist der Error ebenfalls ersichtlich
![error_request_ags_manager](https://github.com/gis-eni/webgis-forum/assets/122986367/e92a476e-1c22-4727-8e65-36602fdcf5eb)

Versuche ich die im Logfile ersichtliche Where clause über die REST-Schnittstelle auszuführen gibt es den gleichen Fehler
![error_rest_query_ags](https://github.com/gis-eni/webgis-forum/assets/122986367/58dcb7aa-fbf3-429b-afb3-3445aa8ab2a9)

Lt. einem Blog-Eintrag von Esri muss bei Feldern vom Type Date eine Date function in die Query eingebaut werden --> https://www.esri.com/arcgis-blog/products/api-rest/data-management/querying-feature-services-date-time-queries/

Wenn ich die Where clause lt. Blogeintrag formatiere [Erhebung_Datum>**DATE** '2015-03-18'] funktioniert die Abfrage in der REST-Schnittstelle und gibt mir die korrekten Objekte zurück
![ok_rest_query_ags](https://github.com/gis-eni/webgis-forum/assets/122986367/783dbecf-2722-4053-930c-7b0491714bd1)

Ist es im Moment möglich nach einem Feld vom Type Date via AGS-REST-Schnittstelle zu suchen?
Falls nicht, könnte das ev. in einer zukünftigen Version eingebaut werden?

Ich habe dir ein Service mit dem Layer zur Verfügung gestellt. Details folgen per Mail. 

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Eine Lösung wird hier erst im Build 5.23.2001 angeboten (bereits verfügbar):

Im CMS gibt es die Möglichkeit bei jedem Suchbegriff optional eine ``FormatExpression`` zu definieren:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/1bfb4220-c385-46eb-b83b-a762f7bf05d0)

 Mit dieser Expression wird die Eingabe formatiert. {0} ist der Platzhalter für die Usereingabe, zB DATE '{0}' => wird zu DATE '2015-5-3'. 
	       
**Achtung:** Wenn eine Expression angeben wird, muss der komplette Ausdruck angegeben inklusive (einfachen) Hochkomma am Anfang oder Ende angeben werden! 
	       zB 'fix_prefix_{0}_fix_postfix'.
